### PR TITLE
Fix the nonlinear bug and add mnist example

### PR DESCRIPTION
**Pull request by Orienfish**
Add one more step in the nonlinear encoding: normalize the weight, which is similar to the random projection case.
To test it, I create a `mnist_nonlinear.py` in the examples.
Without this normalization, the accuracy of running `python mnist_nonlinear.py` is 10.75%.
After adding this normalization, the accuracy becomes 83.93%.

Another example that justify this change is in the `random_projection.py` example:
```
    def encode(self, x):
        enc = self.project(x)
        sample_hv = torch.cos(enc + self.bias) * torch.sin(enc)
        return torchhd.hard_quantize(sample_hv)
```
This encoding function is doing nothing but first a random projection and then passing through the nonlinear encoding.
Since in the random projection, the weight has been normalized; thus in the nonlinear encoding, the weight should also be normalized.

Please let me know if you have any questions or concerns.

### CODE REVIEW

File: examples/mnist_nonlinear.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import torchhd
+from torchhd import embeddings
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+print("Using {} device".format(device))
+
+DIMENSIONS = 10000
+IMG_SIZE = 28
+NUM_LEVELS = 1000
```
Comment (mikeheddes) [inline examples/mnist_nonlinear.py:20]: This variable is not used anymore, could you remove it?

**Review by mikeheddes — APPROVED** [review]
Thank you for the PR and for finding this issue. I verified the accuracy gains that you reported. If you could remove the `NUM_LEVELS` unused variable then I will go ahead and merge you're request.

File: examples/mnist_nonlinear.py (hunk 1 above)
Comment (Orienfish) [reply examples/mnist_nonlinear.py]: Thanks for your quick review! I remove the `NUM_LEVELS` variable as you suggested. I also add one line of comment at the beginning of `mnist_nonlinear.py` example.

**Comment by mikeheddes** [comment]
Congrats on your first contribution to the library! Thank you for your help :)
